app/javascript/mastodon/features/compose/components/search.tsx

```diff
@@ -263,113 +360,63 @@ export const Search: React.FC<{
         const couldBeURL =
           trimmedValue.startsWith('https://') && !trimmedValue.includes(' ');
 
+        let mastoPath;
         if (couldBeURL) {
-          newQuickActions.push({
-            key: 'open-url',
-            label: (
-              <FormattedMessage
-                id='search.quick_action.open_url'
-                defaultMessage='Open URL in Mastodon'
-              />
-            ),
-            action: async () => {
-              const result = await dispatch(openURL({ url: trimmedValue }));
-
-              if (isFulfilled(result)) {
-                if (result.payload.accounts[0]) {
-                  history.push(`/@${result.payload.accounts[0].acct}`);
-                } else if (result.payload.statuses[0]) {
-                  history.push(
-                    `/@${result.payload.statuses[0].account.acct}/${result.payload.statuses[0].id}`,
-                  );
-                }
-              }
+          newQuickActions.push(QuickActionGenerators.couldBeURL(trimmedValue));
 
-              unfocus();
-            },
-          });
+          // presume URL is from a mastodon server; not just some random web URL
```

Comment:
Given the fediverse is very diverse, I am not a big fan of this presumption. I'd prefer to either leave this logic out or better isolate it from the rest.